Additional vc_asserts

vulcandth opened the pull request:
This PR adds additional Virtual Console build asserts.

Please reference #892 and Rangi42/polishedcrystal#684

Resolves #892

Rangi42 commented on engine/menus/save.asm:
```diff
@@ -170,6 +170,8 @@ AddHallOfFameEntry:
 		"sMobileEventIndex is no longer located at 01:be3c."
 	vc_assert BANK(sMobileEventIndexBackup) == $1 && sMobileEventIndexBackup == $be44, \
 		"sMobileEventIndexBackup is no longer located at 01:be44."
+	vc_assert MOBILE_EVENT_OBJECT_GS_BALL == $B, \
+		"MOBILE_EVENT_OBJECT_GS_BALL is no longer equal to $B."
```
```suggestion
	vc_assert MOBILE_EVENT_OBJECT_GS_BALL == $0b, \
		"MOBILE_EVENT_OBJECT_GS_BALL is no longer equal to $0b."
```

Rangi42 commented on engine/link/link.asm:
```diff
@@ -2311,6 +2311,14 @@ WaitForLinkedFriend:
 	ldh [rSC], a
 	ld a, (1 << rSC_ON) | (0 << rSC_CLOCK)
 	vc_hook linkCable_fake_begin
```
Add a comment above this `vc_hook` like the Celebi one, explaining briefly how it uses these values.

vulcandth commented:
Going to do a little bit extra testing real quick to see If I can't be a bit more specific on what the linkCable\_fake hooks are doing.

vulcandth commented:
Removed two of the vc_assert's that I determined were not being used through testing. Added the one comment you requested. 

Further patch Identification and Comments will be done in the other PR.

Rangi42 commented on engine/link/link.asm:
```diff
@@ -2310,7 +2310,14 @@ WaitForLinkedFriend:
 	ld a, (0 << rSC_ON) | (0 << rSC_CLOCK)
 	ldh [rSC], a
 	ld a, (1 << rSC_ON) | (0 << rSC_CLOCK)
+; This vc_hook causes the virtual console to set [hSerialConnectionStatus] to
+; [USING_INTERNAL_CLOCK] ($02) which allows the player to proceed past the
+; Link Receptionists' "Please Wait.".
```
```suggestion
; This vc_hook causes the Virtual Console to set [hSerialConnectionStatus] to
; USING_INTERNAL_CLOCK, which allows the player to proceed past the
; Link Receptionist's "Please wait."
```

Rangi42 commented on engine/menus/save.asm:
```diff
@@ -170,6 +170,8 @@ AddHallOfFameEntry:
 	vc_assert BANK(sMobileEventIndexBackup) == $1 && sMobileEventIndexBackup == $be44, \
 		"sMobileEventIndexBackup is no longer located at 01:be44."
+	vc_assert MOBILE_EVENT_OBJECT_GS_BALL == $0b, \
+		"MOBILE_EVENT_OBJECT_GS_BALL is no longer equal to $0b."
```
Please remove the "`($b)`" in the comment.

Rangi42 commented:
Thank you! pokegold will need the `hSerialConnectionStatus` one too.
